src/components/buttons/button/styles.ts

```diff
@@ -0,0 +1,114 @@
+import styled, { css, CssWithTheme } from 'styled-components'
+import { StyledButtonProps } from './index'
+import { ButtonSize } from './enums'
+
+const { Normal } = ButtonSize
+
+function getColorStyle({
```

Comment:
Mám tady dvě otázky:
1) Když je button "inverted", tak nemá mít styly pro hover?
2) Moc se mi nelíbí to posílání "flagů" - jednak to nedodržuje to, že by funkce měla dělat jednu věc, ale hlavně mám obavu, že v budoucnu by ta komplexita celé funkce mohla vyrůst (přidají se další flagy, pokrytí dalších případů atd.), tak bych se zkusil zamyslet, jestli by to nešlo rozbít do 3 menších funkcí :)

Reply:
@patrikbraborec, díky moc za review. 🙏  ... Házím souhrný komentář tu. 

1. inverted opravdu barvy nemění při hover, jen box shadow
2. Zamyslet se nad level of abstraction byl dobrý point, díky za něj. Tam, kde větvení bylo komplexní jsem upravil. Tam, kde se jednalo o jednu podmínku, jsem zanechal. Trochu jsem váhal nad velikostí souboru po této úpravě, ale je to hraniční, pokud naroste ještě více, vyřešíme v rámci těch přídavků. 

Historii jsem vyčistil, ponechám nějaký prostor pro last-minute připomínky a poté zamerguji. 

Díky všem za pomoc. 👍

Reply:
Super :) jinak za mě je to ok, takže můžeme mergovat kdyžtak.. :)